src/main/java/com/gojek/beast/worker/OffsetCommitWorker.java

```diff
@@ -1,84 +1,120 @@
+                            OffsetMetadata previousOffset = (OffsetMetadata) partitionsCommitOffset.getOrDefault(topicPartition, new OffsetMetadata(Integer.MIN_VALUE));
+                            OffsetMetadata newOffset = new OffsetMetadata(offsetAndMetadata.offset());
+                            if (previousOffset.compareTo(newOffset) < 0) {
+                                partitionsCommitOffset.put(topicPartition, newOffset);
+                            }
+                        });
+                        offsetState.resetOffset();
+                        offsetClubbedBatches++;
+                        break;
+                    } else {
+                        if (offsetState.shouldCloseConsumer(partitionsCommitOffset)) {
+                            statsClient.increment("committer.ack.timeout");
+                            return new FailureStatus(new RuntimeException("Acknowledgement Timeout exceeded: " + offsetState.getAcknowledgeTimeoutMs()));
+                        }
+
+                        log.debug("waiting for {} acknowledgement for offset {}, {}, {}", defaultSleepMs, currentOffset, offsetState.partitionOffsetAckSize(), currentOffset);
+                        sleep(defaultSleepMs);
+                    }
+                }
+                statsClient.timeIt("committer.queue.wait.ms", commitQueuePollStartTime);
             }
-            statsClient.timeIt("committer.processing.time", start);
+            commit(partitionsCommitOffset);
+
+            statsClient.gauge("committer.clubbed.offsets", offsetClubbedBatches);
+            statsClient.timeIt("committer.processing.time", startTime);
         } catch (InterruptedException | RuntimeException e) {
+            log.info("Received {} exception: {}, resetting committer", e.getClass(), e.getMessage());
+            e.printStackTrace();
             return new FailureStatus(new RuntimeException("Exception in offset committer: " + e.getMessage()));
         }
         return new SuccessStatus();
     }
 
-    private void commit() {
-        Map<TopicPartition, OffsetAndMetadata> partitionsCommitOffset = commitQueue.remove().getPartitionsCommitOffset();
-        kafkaCommitter.commitSync(partitionsCommitOffset);
-        partitionOffsetAck.remove(partitionsCommitOffset);
-        Records nextOffset = commitQueue.peek();
-        if (nextOffset != null) offsetState.resetOffset(nextOffset.getPartitionsCommitOffset());
-        log.info("commit partition {} size {}", partitionsCommitOffset.toString(), partitionsCommitOffset.size());
+    private void commit(Map<TopicPartition, OffsetAndMetadata> partitionsCommitOffset) {
+        if (partitionsCommitOffset.size() != 0) {
+            kafkaCommitter.commitSync(partitionsCommitOffset);
+        }
+        log.info("committed offsets partition {} size {}", partitionsCommitOffset.toString(), partitionsCommitOffset.size());
```

Comment:
Is this log really helpful? Can we mark this either debug or remove serializing the whole map to a string?

Reply:
Since we will be committing after batching commits, we want to log when the offsets were committed, there is no other info log in the OffsetCommitter

Reply:
@mauliksoneji  Yeah, that's fine, I am just wondering what's the use of seeing offset values per partition in the log, we can just log the size of that map?

Reply:
This doesn't have much information, the maximum number of keys would be the topic-partition subscribed by the pod, and there is only max offset in the value, this is not that much information.